recording-daemon/mix.c

```diff
@@ -82,32 +90,46 @@ static void mix_input_reset(mix_t *mix, unsigned int idx) {
 	mix->in_pts[idx] = 0;
 }
 
+unsigned int mix_get_index(mix_t *mix, void *ptr, unsigned int stream_channel_slot) {
+
+	ilog(LOG_DEBUG, "next index is set to %u for slot %u. channel slots for this mix are %u", mix->next_idx[stream_channel_slot], stream_channel_slot, mix->channel_slots);
+
+	if(mix->next_idx[stream_channel_slot] == mix_num_inputs+1) {
```

Comment:
I think validation that `stream_channel_slot` is actually `< MIX_MAX_INPUTS` is missing? AFAICT there's nothing preventing somebody from setting `recording-media-slots=500` or something.